src/main.ts

```diff
@@ -16,48 +21,101 @@ async function commandExists(cmd: string): Promise<boolean> {
   return code === 0
 }
 
+async function runSbt(baseDir: string, pluginVersionInput: string): Promise<void> {
+  const projectDir = path.join(baseDir, 'project')
+
+  if (!fs.existsSync(projectDir)) {
+    core.setFailed(`${baseDir} is not a valid sbt project: missing folder '${projectDir}'.`)
+    return
+  }
+
+  const uuid = crypto.randomUUID()
+  const pluginFile = path.join(projectDir, `github-dependency-graph-${uuid}.sbt`)
+
+  const pluginVersion =
+    pluginVersionInput.length === 0 ? defaultSbtPluginVersion : pluginVersionInput
+
+  const pluginDep = `addSbtPlugin("ch.epfl.scala" % "sbt-github-dependency-graph" % "${pluginVersion}")`
+  await fsPromises.writeFile(pluginFile, pluginDep)
+
+  const sbtExists = await commandExists('sbt')
+  if (!sbtExists) {
+    core.setFailed('Not found sbt command')
+    return
+  }
+
+  const input = {
+    projects: core
+      .getInput('projects')
+      .split(' ')
+      .filter(value => value.length > 0),
+    scalaVersions: core
+      .getInput('scala-versions')
+      .split(' ')
+      .filter(value => value.length > 0),
+  }
```

Comment:
This is still one of the parts that I'm really unsure about. In the Mill plugin I currently don't support any filtering. Can you explain a bit more what you think the best use case is to filter _in_ modules? I can maybe see if there is a module that is problematic or a module that is only used for local testing or something that someone would want to filter out, but I'm a bit unsure why someone would want to only include certain modules.

Either way, I can support this in Mill, it's just I don't want to unless we know people are going to use it. I saw for example in [Metals](https://github.com/scalameta/metals/blob/main/.github/workflows/sbt-dependency-graph.yml#L14) you only enabled it for 3 versions, but we do publish for every Scala version in mtags, and in theory we could be including different versions of dependencies for different versions of Scala. How come you only enabled 3? Also, in the other repos that you've added the sbt plugin to so far, did you enable it for everything, or are you also using filtering?

Reply:
Yes I agree this is very questionable. For the module part, it is true that people should not want to exclude any module except if there is a problem with one module. So filtering out is more useful than filtering in.

On the Scala version part: if there is a project that is cross-compiled on all patch versions of Scala, for instance mtags, it is very likely that all groups of minor versions have the same set of dependencies. So there is a lot of redundancy in the snapshot and maybe we should not publish all manifests but only one manifest per minor Scala version. One for 2.12, one for 2.13 and one for 3. That was the initial idea. But what if we publish all manifests on all patch versions? It will be slower and the snapshot would be bigger. Not a big deal. So maybe the filtering in on Scala version is not necessary.

In the sbt project which is cross-compiled on Scala 2.12.16 and 2.13.8, the resolution on 2.13.8 fails for some reason. So here it is useful to filter out a Scala version. But again, filtering out is more useful.

There is another question then: should we filter out a Scala version on all modules or only on a few modules. In the case of sbt I only need to filter out one module on 2.13.8.

So maybe we should rework this API completely. Can we find an API that works well on sbt and mill?

Something like this maybe:
```yaml
## Ignore module foo on Scala version 2.13, module bar on all Scala versions and all modules on Scala version 2.12
modules-ignore: foo_2.13 bar_* *_2.12
```

That works also well with Scala platforms:
```yaml
## Ignore all Scala.js 0.6 and Scala Native 0.4 modules
modules-ignore: *_sjs0.6_* *_native0.4_*
```

This syntax with `*` is the [Glob syntax](https://en.wikipedia.org/wiki/Glob_(programming)) also called the [filter pattern syntax](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet) in Github.

Reply:
If you call `artifactName` on a `JavaModule` you should have something like `foo` for pure Java modules, `foo_2.13` for Scala 2.13 modules and `foo_sjs1_2.13` for Scala.js 1.x modules.

Reply:
Actually it is not `artifactName` but `artifactId`.

Reply:
> If you call artifactName on a JavaModule you should have something like foo for pure Java modules, foo_2.13 for Scala 2.13 modules and foo_sjs1_2.13 for Scala.js 1.x modules.

Ahh, good point. I guess that would work then.

Reply:
Out of curiosity, what would you use to implementing the filtering by globbing, since you can't really just use `scala.util.matching.Regex`, which is what my mind first went to since globs are different?